src/Nethermind/Nethermind.Serialization.Ssz/Ssz.ShardBlobTransaction.cs

```diff
@@ -0,0 +1,322 @@
+using System.Runtime.CompilerServices;
+using Nethermind.Core;
+using Nethermind.Core.Crypto;
+using Nethermind.Core.Eip2930;
+using Nethermind.Int256;
+
+namespace Nethermind.Serialization.Ssz;
+
+public static partial class Ssz
+{
+    public static int BlobTransactionNetworkWrapperLength(Transaction container)
+    {
+        ShardBlobNetworkWrapper? wrapper = container.NetworkWrapper as ShardBlobNetworkWrapper;
+        if (wrapper is null)
+        {
+            throw new ArgumentException("Wrapper should present in network form", nameof(container.NetworkWrapper));
+        }
+
+        int transactionDynamicOffset = 4 * VarOffsetSize;
+        int commitmentsDynamicOffset = transactionDynamicOffset + SignedBlobTransactionLength(container);
+        int blobsDynamicOffset =
+            commitmentsDynamicOffset + wrapper.Commitments.Length;
+        int proofsDynamicOffset = blobsDynamicOffset + wrapper.Blobs.Length;
+
+        return proofsDynamicOffset + wrapper.Proofs.Length;
+    }
+
+    public static int SignedBlobTransactionLength(Transaction container) =>
+        TransactionDynamicOffset + TransactionLength(container);
+
+    public static int TransactionLength(Transaction transaction)
+    {
+        int toDynamicOffset = 192;
+        int dataDynamicOffset = toDynamicOffset + OptionalAddressLength(transaction.To);
+        int accessListDynamicOffset = dataDynamicOffset + (transaction.Data?.Length ?? 0);
+        int blobVersionedHashesToDynamicOffset = accessListDynamicOffset;
+        if (transaction.AccessList is not null)
+        {
+            foreach (KeyValuePair<Address, IReadOnlySet<UInt256>> pair in transaction.AccessList.Data)
+            {
+                blobVersionedHashesToDynamicOffset +=
+                    VarOffsetSize + Address.ByteLength + VarOffsetSize + pair.Value.Count * 32;
+            }
+        }
+
+        return blobVersionedHashesToDynamicOffset + (transaction.BlobVersionedHashes?.Length ?? 0) * 32;
+    }
+
+    private static int OptionalAddressLength(Address? address)
+    {
+        return sizeof(byte) + (address is null ? 0 : Address.ByteLength);
+    }
+
+    private const int SignatureLength = 65;
+
+    private const int TransactionDynamicOffset = SignatureLength + 4;
+
+    public static void EncodeSignedWrapper(Span<byte> span, Transaction transaction)
+    {
+        ShardBlobNetworkWrapper? wrapper = transaction.NetworkWrapper as ShardBlobNetworkWrapper;
+        if (wrapper is null)
+        {
+            throw new ArgumentException("Wrapper should present in network form", nameof(transaction.NetworkWrapper));
+        }
+
+        int transactionDynamicOffset = 4 * VarOffsetSize;
+        int commitmentsDynamicOffset = transactionDynamicOffset + SignedBlobTransactionLength(transaction);
+        int blobsDynamicOffset =
+            commitmentsDynamicOffset + wrapper.Commitments.Length;
+        int proofsDynamicOffset = blobsDynamicOffset + wrapper.Blobs.Length;
```

Comment:
Ok, here we need all these variables, and in `BlobTransactionNetworkWrapperLength` it was probably copy pasted, so that's fine to keep it in the same form. Nevermind